GiniVision/Classes/Core/Multipage Review/MultipageReviewPagesCollectionFooter.swift

```diff
@@ -0,0 +1,153 @@
+//  GiniVision
+//
+//  Created by Enrique del Pozo Gómez on 4/26/18.
+//
+
+import Foundation
+
+final class MultipageReviewPagesCollectionFooter: UICollectionReusableView {
+    
+    static let identifier = "MultipageReviewPagesCollectionFooterIdentifier"
+    var didTapAddButton: (() -> Void)?
+    
+    fileprivate static let contentSize = MultipageReviewPagesCollectionCell.size
+    var trailingConstraint: NSLayoutConstraint?
+    
+    fileprivate lazy var roundMask: UIView = {
+        let view = UIView(frame: .zero)
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.backgroundColor = .gray
+        view.layer.cornerRadius = 5.0
+        view.layer.masksToBounds = true
+        return view
+    }()
+    fileprivate lazy var addButton: UIButton = {
+        let button = UIButton(type: .custom)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.setImage(UIImageNamedPreferred(named: "addCircleIcon"), for: .normal)
+        button.tintColor = .white
+        button.addTarget(self, action: #selector(addImageButtonAction), for: .touchUpInside)
+        return button
+    }()
+    fileprivate lazy var addLabel: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.text = NSLocalizedString("ginivision.multipagereview.addButtonLabel",
+                                       bundle: Bundle(for: GiniVision.self),
+                                       comment: "label shown below add button")
+        label.textAlignment = .center
+        label.font = UIFont.boldSystemFont(ofSize: 12)
+        label.numberOfLines = 0
+        label.textColor = .white
+        return label
+    }()
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        addSubview(roundMask)
+        roundMask.addSubview(addButton)
+        roundMask.addSubview(addLabel)
+        
+        addInnerShadow()
+        addConstraints()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    class func size(in collectionView: UICollectionView) -> CGSize {
+        let padding = self.padding(in: collectionView)
+        let height = MultipageReviewPagesCollectionFooter.contentSize.height +
+            padding.top +
+            padding.bottom
+        let width = MultipageReviewPagesCollectionFooter.contentSize.width +
+            padding.left +
+            padding.right
+        
+        return CGSize(width: width, height: height)
+    }
+    
+    class func padding(in collectionView: UICollectionView? = nil) -> UIEdgeInsets {
+        var rightPadding: CGFloat = 10
+        if let collection =  collectionView {
+            rightPadding = ((collection.frame.width -
+                MultipageReviewPagesCollectionCell.size.width) / 2)
+        }
+        
+        return UIEdgeInsets(top: 10, left: 10, bottom: 10, right: rightPadding)
```

Comment:
All these tens displease me. I'd use `defaultPadding = 10` and have a `rightPadding` that's calculated.